src/test/ts/part3/Part3Ex2Test.ts

```diff
@@ -1,65 +1,114 @@
+    your node is a Text, then the offset is going to tell you how many
+    characters into the text your selection endpoint is.
+
+    This concept is summed up really nicely by the builtin "Range" API exported
+    by the browser. Let's create a Range object, and make sure it works.
      */
-    const editor = hook.editor();
+    // TinyMCE also exposes this document.createRange() through the `dom` API
+    // this ensures the range object is using the same document as the editor
+    const range = editor.dom.createRng();
 
-    // Now what?
-    // Let's just make sure that the bold button we asked for is there
+    const contentBody = TinyDom.body(editor);
+    // Let's get the <p>And some <strong>bolded</strong> content</p>
+    const paragraph = Traverse.child(contentBody, 1).getOrDie('Unable to find second child of editor body');
+    // And then get the <strong>bolded</strong> (child 0 would be the text node "And some ")
+    const strong = Traverse.child(paragraph, 1).getOrDie('Unable to find second child of paragraph');
+    // And finally let's get the text node "bolded"
+    const text = Traverse.child(strong, 0).getOrDie('Unable to find text inside strong element');
 
-    // the element containing the editor and all of its UI (TinyDom is another
-    // mcagar API, this one gets SugarElements out of editor things).
-    const container = TinyDom.container(editor);
+    range.setStart(text.dom, 0);
+    range.setEnd(text.dom, 'bolded'.length);
 
-    // assert a bold button is in the container
-    // This is from the agar library, as it's a generic utility that isn't
-    // strictly editor-related
-    UiFinder.exists(container, 'button[title="Bold"]');
+    editor.selection.setRng(range);
+
+    assert.equal(editor.selection.getContent(), 'bolded');
   });
 
-  it('has content like an editor', () => {
+  it('selects content like an editor, take II', () => {
     const editor = hook.editor();
+    /*
+    Now in the last test, we put together a block of code that's a really good
+    teaching opportunity for another mcagar API. Here's the code in question
 
-    editor.setContent(/* TODO */ "<p>Hello world</p>");
+    const contentBody = TinyDom.body(editor);
+    // Let's get the <p>And some <strong>bolded</strong> content</p>
+    const paragraph = Traverse.child(contentBody, 1).getOrDie('Unable to find second child of editor body');
+    // And then get the <strong>bolded</strong> (child 0 would be the text node "And some ")
+    const strong = Traverse.child(paragraph, 1).getOrDie('Unable to find second child of paragraph');
+    // And finally let's get the text node "bolded"
+    const text = Traverse.child(strong, 0).getOrDie('Unable to find text inside strong element');
 
+    And here's how you'd do all of this with an agar API.
+     */
+    const contentBody = TinyDom.body(editor);
+    const text = Cursors.follow(contentBody, [ 1, 1, 0 ]).getOrDie();
```

Comment:
Does `Cursors.follow` allow for offsets, or is it purely the dom nodes? Looks like the latter? Should we make a quick note of that?
```suggestion
    const text = Cursors.follow(contentBody, [ 1, 1, 0 ]).getOrDie();
    // Note that Cursors.follow only deals with DOM nodes, not offsets, and it uses the same node
    // for both the selection start and end node
```